torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1401,6 +1414,18 @@ def sample_inputs_fliplr_flipud(op_info, device, dtype, requires_grad):
                        SkipInfo('TestCommon', 'test_variant_consistency_jit',
                                 device_type='cuda', dtypes=[torch.float16]),
                    )),
+    OpInfo('std',
+           dtypes=floating_types_and(),
+           dtypesIfCUDA=floating_and_complex_types_and(torch.half, torch.bfloat16),
+           sample_inputs_func=sample_inputs_std_var,
+           supports_tensor_out=False,
+           test_complex_grad=False,
+           test_inplace_grad=False,
+           skips=(SkipInfo('TestGradients', 'test_fn_gradgrad'),
```

Comment:
Would you elaborate on these two skips?

Reply:
I removed the first skip, after reviewing it I realised the tests passed as-is. (probably fixed by the `test_complex_grad` changes here). As for the second skip, the CPU version of `std` and `var` have only partial support for half and complex, because they sometimes call into the old TH version for all-reductions:

https://github.com/pytorch/pytorch/blob/ac0a3cc5fdaf2b559d4b96f41d7accdc5fe9c320/aten/src/ATen/native/ReduceOps.cpp#L1151-L1156

So for the `OpInfo`: `Half`, `ComplexFloat` and `ComplexDouble` are declared as unsupported on cpu but not all sample inputs will raise.

Reply:
That makes sense. Would you file an issue and add a comment citing it?

Reply:
Opened #51127